js/components/plates.js

```diff
@@ -362,7 +370,9 @@ export default class Plates extends PureComponent {
         </div>
         {
           authoring &&
-          <Authoring loadModel={this.loadModel} unloadModel={this.unloadModel} setOption={this.handleOptionChange} />
+          <Authoring loadModel={this.loadModel} unloadModel={this.unloadModel}
+            setDensities={this.setDensities} setOption={this.handleOptionChange}
+            plates={this.modelProxy.plates} />
```

Comment:
We shouldn't do that in React. Props should be based only on parent component's state and props only. If `modelProxy.plates` is changed, React won't know that it should trigger re-rendering. I know that in this particular case, probably it won't happen. Or re-rendering will be triggered by some other property update. But in general, such approach can lead to bugs that are pretty hard to detect.

Also, I think we have too much state related to plate densities in general:
1. This property
2. intermediateDensities in Authoring
3. internal state of SortableDensities component

I think it could be reduced to one state kept at the top level. What I'd do:
- add `plateDensities` to `Plates` (this component) React state.
- `platesDensity` state is updated when there's a new message from the model worker, so it's always kept in sync with the model.
- pass it down to Authoring and then down to SortableDensities.
- when sorting happens, SortableDenisties triggers a callback that updates `platesDensity` here.
- update of `plateDensities` here triggers postMessage to the model worker. Make sure that this message is sent only if there's any difference in densities, so we avoid an infinite cycle of messages between this component and model worker.

I know it's more typing and seems more complicated, but it feels to me that it's a correct approach without shortcuts. And if we follow this pattern for new things in the future, hopefully, it'll save some debugging time.